README.md

```diff
@@ -223,6 +228,47 @@ Supported domains: BLAS, LAPACK, RNG
     </tbody>
 </table>
 
+### Supported Configurations with hipSYCL:
```

Comment:
I am not sure if this should be part of this PR, but shouldn't the intel LLVM compiler and the dpc++ compiler be handled differently here?

Reply:
Thanks!

Reply:
Agree, we do have a difference in support for dpc++ and llvm compilers that we didn't reflect here. Maybe to avoid the duplication in the multiple tables what do you think about adding one more column `Supported compilers`  and have only one table? So e.g. for `Nvidia cuBLAS`  we can list LLVM and hipSYCL compilers in this column.

Reply:
We have not tested cuBLAS backend with hipSYCL locally yet. Could you save it for a future PR?

Reply:
I also added a extra table to clarify the current domain support

Reply:
Thanks for tables in the build section!
For this one I still think it will be easier to read as one table instead of two. For some reasons I can't open PR with the suggested table to your branch.

So here is the table in the row format

```
### Supported Configurations:

Supported domains: BLAS, LAPACK, RNG

#### Linux*

<table>
    <thead>
        <tr align="center" >
            <th>Domain</th>
            <th>Backend</th>
            <th>Library</th>
            <th>Supported Link Type</th>
            <th>Supported Compiler</th>		
        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan=5 align="center">BLAS</td>
            <td align="center">x86 CPU</td>
            <td rowspan=2 align="center">Intel(R) oneAPI Math Kernel Library</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++, LLVM*, hipSYCL</td>
        </tr>
        <tr>
            <td align="center">Intel GPU</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">NVIDIA GPU</td>
            <td align="center">NVIDIA cuBLAS</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">LLVM*, hipSYCL</td>
        </tr>
        <tr>
            <td align="center">x86 CPU</td>
            <td align="center">NETLIB LAPACK</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++, LLVM*, hipSYCL</td>
        </tr>
	    <tr >
            <td align="center">AMD GPU</td>
            <td align="center">AMD rocBLAS </td>
            <td align="center">Dynamic, Static</td>
            <td align="center">hipSYCL</td>
        </tr>
        <tr>
            <td rowspan=3 align="center">LAPACK</td>
            <td align="center">x86 CPU</td>
            <td rowspan=2 align="center">Intel(R) oneAPI Math Kernel Library</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++, LLVM*</td>
        </tr>
        <tr>
            <td align="center">Intel GPU</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">NVIDIA GPU</td>
            <td align="center">NVIDIA cuSOLVER</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">LLVM*</td>
        </tr>
        <tr>
            <td rowspan=3 align="center">RNG</td>
            <td align="center">x86 CPU</td>
            <td rowspan=2 align="center">Intel(R) oneAPI Math Kernel Library</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++, LLVM*</td>
        </tr>
        <tr>
            <td align="center">Intel GPU</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">NVIDIA GPU</td>
            <td align="center">NVIDIA cuRAND</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">LLVM*</td>
        </tr>
    </tbody>
</table>

\* LLVM - [Intel project for LLVM* technology](https://github.com/intel/llvm) with support for [NVIDIA CUDA](https://intel.github.io/llvm-docs/GetStartedGuide.html#build-dpc-toolchain-with-support-for-nvidia-cuda)

#### Windows*

<table>
    <thead>
        <tr align="center" >
            <th>Domain</th>
            <th>Backend</th>
            <th>Library</th>
            <th>Supported Link Type</th>
            <th>Supported Compiler</th>	
        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan=3 align="center">BLAS</td>
            <td align="center">x86 CPU</td>
            <td rowspan=2 align="center">Intel(R) oneAPI Math Kernel Library</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">Intel GPU</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">x86 CPU</td>
            <td align="center">NETLIB LAPACK</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td rowspan=2 align="center">LAPACK</td>
            <td align="center">x86 CPU</td>
            <td rowspan=2 align="center">Intel(R) oneAPI Math Kernel Library</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">Intel GPU</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
        <tr>
            <td align="center">RNG</td>
            <td align="center">x86 CPU</td>
            <td align="center">Intel(R) oneAPI Math Kernel Library</td>
            <td align="center">Dynamic, Static</td>
            <td align="center">DPC++</td>
        </tr>
    </tbody>
</table>
```
![Capture](https://user-images.githubusercontent.com/28574953/153960195-b850d94b-8be4-4a36-8842-20779f025597.JPG)

Reply:
@sbalint98 I missed that we had these tables. I think these tables cover the three new tables we just added. If you agree, can you please remove those? Very sorry for the inconvenience. 

Also, for this table, can we remove the hipSYCL from BLAS/NVIDIA GPU/NVIDIA cuBLAS row? (not tested)

Reply:
@mmeterel Can you elaborate on the rationale behind not just giving the hipSYCL cuBLAS backend a quick test run? It's not clear to me why we wouldn't advertise something that we know works and that we support. hipSYCL is often misunderstood as having a particular focus on AMD hardware, when it is in fact far more general and supports other hardware like NVIDIA GPUs just as well. We would like to prevent this impression.

Is this purely related to the organization of PRs and retaining a rocBLAS focus in this PR?

Reply:
@illuhad There are mainly three reasons why I would prefer a two step approach for adding cuBLAS support for hipSYCL.

1. The first one is the organization of PRs as you mentioned. As the PR title says, this is about adding rocBLAS backend. 
2. Second, as you have seen in our offline discussions, correct testing of rocBLAS + hipSYCL locally took some time. For cuBLAS + hipSYCL, we would like to first validate it locally - which might take a while and currently don't have BW. We also need to discuss internally how/if we are going to support it in CI and make a plan for it. 

Otherwise, we are very excited to add hipSYCL support into our project and looking forward to extend its capabilities. Is there another PR for Intel GPUs coming soon as well? :)

Reply:
@mmeterel Thank you for the explanation. I believe your existing hipSYCL installation might already be able to target CUDA. You'd only have to point cmake to CUDA when building hipSYCL if CUDA is installed in a non-standard location :-)

I think for Intel GPUs we will need some more time, as the hipSYCL level zero backend does not yet support backend interoperability to the same extent as the CUDA or HIP backends. So, oneMKL+hipSYCL+Intel GPUs is a bit further in the future, but it would be great to have support for this at some point :)

@sbalint98 Can you move the addition of hipSYCL cuBLAS to the documentation to a separate PR?

Reply:
@mkrainiuk @mmeterel Thank you for the changes, I applied them, let me know if there is still anything else to change. 

@illuhad I'll create a PR as soon as this is in ;)

Reply:
Thanks @sbalint98! I think we no longer need https://github.com/sbalint98/oneMKL/tree/ustream-add-rocm#supported-configurations-with-hipsycl, could you please remove it?

Reply:
And I think we discussed enabling hipSYCL in the future PR after full testing <html>
<body>
<!--StartFragment-->

BLAS | NVIDIA cuBLAS | Dynamic, Static | LLVM*<s>, hipSYCL</s>
-- | -- | -- | --


<!--EndFragment-->
</body>
</html>

@mmeterel please correct me if I'm wrong.

Reply:
> We have not tested cuBLAS backend with hipSYCL locally yet. Could you save it for a future PR?

@sbalint98 I still see hipSYCL for cuBLAS row (unless I am looking at an old page). Could you please double check?

Reply:
my bad, sorry, now removed